### PR TITLE
Fix coverity hits

### PR DESCRIPTION
**Pull request by md-shafiul-alam**
# Description
- Removed unreachable return statements
- Refactored unreachable branches
- Added checks for non-zero division
- Removed redundant comparisons

### CODE REVIEW

File: cpp/daal/src/algorithms/pca/transform/pca_transform_dense_default_batch_impl.i [hunk 1]
```diff
@@ -105,10 +105,6 @@ services::Status TransformKernel<algorithmFPType, method, cpu>::compute(NumericT
     const algorithmFPType * pBasis = basis.get();
 
     size_t numRowsInBlock = _numRowsInBlock;
-    if (numRowsInBlock < 1)
```
Comment (ethanglaser) [inline cpp/daal/src/algorithms/pca/transform/pca_transform_dense_default_batch_impl.i:108]: would we want to replace this with an assertion/check?

**Comment by ethanglaser** [comment]
Please run clang script

**Review by ethanglaser — APPROVED** [review]
Looks good to me, will need manager approval as well.

File: cpp/daal/src/algorithms/pca/transform/pca_transform_dense_default_batch_impl.i (hunk 1 above)
Comment (md-shafiul-alam) [reply cpp/daal/src/algorithms/pca/transform/pca_transform_dense_default_batch_impl.i:108]: In this case, _numRowsInBlock is assigned internally in a header file. We do not expect it to be assigned to zero or negatives.

**Pull request merged**